pandas/tests/reshape/concat/test_concat.py

```diff
@@ -698,3 +698,25 @@ def test_concat_posargs_deprecation():
         result = concat([df, df2], 0)
     expected = DataFrame([[1, 2, 3], [4, 5, 6]], index=["a", "b"])
     tm.assert_frame_equal(result, expected)
+
+
+def test_concat_retain_attrs():
+    # GH#41828
+    d = {'col1': [1, 2], 'col2': [3, 4]}
+    df1 = pd.DataFrame(data=d)
+    df1.attrs = {1: 1}
+    df2 = pd.DataFrame(data=d)
+    df2.attrs = {1: 1}
+    df = pd.concat([df1, df2])
+    assert df.attrs == {1: 1}
```

Comment:
construct the appropraiate expected value then use `tm.assert_frame_equal`, you can *also* directly compare the attrs for clarity